### PR TITLE
Shell script cleanups

### PR DESCRIPTION
**Pull request by practicalswift**
Shell script cleanups:
* Add required space to `[ -n ]`.
* Avoid quote within quote.
* Exit if `cd` fails.
* Remove `\n` which is not handled by `echo`.
* ~~Remove redundant `$` in arithmetic variable expression.~~
* ~~Use `$(command)` instead of legacy form `` `command` ``.~~
* Arrays are not supported in POSIX `sh`. Use `bash` when arrays are used.
* ~~`[ foo -a bar ]` is not well defined, use `[ foo ] && [ bar ]` instead.~~
* ~~`[ foo -o bar ]` is not well defined, use `[ foo ] || [ bar ]` instead.~~

### CODE REVIEW

**Comment by gmaxwell** [comment]
do we have someone on OpenBSD or likewise to test this?

**Comment by MarcoFalke** [comment]
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

**Comment by practicalswift** [comment]
@MarcoFalke Fixed! :-)

**Comment by laanwj** [comment]
> do we have someone on OpenBSD or likewise to test this?

We don't have any tests for most of these scripts, period :(

**Comment by TheBlueMatt** [comment]
With the exception of the sh -> bash change when needed and clear fixes for edge cases (eg the || exit 1s), both of which I think we should do, I'm not sure what this accomplishes aside from code churn...are the other things going to break on some shell, or are they deprecated in a way that they will be removed within the next ~decade? We've had plenty of issues with otherwise innocuous shell script changes breaking dash-based systems, not to mention BSD incompatibilities, plus now WSL, so there is nontrivial review for some of these things, if its just because they are prettier or feel nicer, I really think we should cut back some of these changes.

**Comment by practicalswift** [comment]
@TheBlueMatt Thanks for reviewing.

I've now removed the changes that could be claimed to be in the "deprecated but not likely to be removed any time soon" and the "feels prettier/nicer" categories.

I've also unsquashed the changes so that this PR can be reviewed on a more detailed per-change/commit level. The requested squashing was perhaps a bit premature. I'll squash again when the only ACK:ed changes remain :-)

The remaining changes are:

1. **Remove unused variables**
Rationale: Obvious mistake corrected.

2. **Remove "\n" from echo argument. echo does not support escape sequences.**
Rationale: Obvious mistake corrected - `echo "\n"` prints a backslash, the letter n and a newline.

3. **Add error handling: exit if cd fails**
Rationale: Obvious mistake corrected - missing error handling.

4. **Add required space to [[ -n "$1" ]] (previously [[ -n"$1" ]])**
Rationale: Obvious mistake corrected - the syntax is `[[ -n STRING ]]`, not `[[ -nSTRING ]]`.

5. **Fix incorrect quoting of quotes (the previous quotes had no effect beyond unquoting)**
Rationale: `echo "1 "2" 3"` prints `1 2 3` and not `1 "2" 3`. Change to `echo "1 \"2\" 3"` where quotes are expected in the output.

6. **Use bash instead of POSIX sh. POSIX sh does not support arrays.**
Rationale: Assumptions are better stated explicitly than implicitly :-)

@TheBlueMatt How do you feel about the remaining changes? All of those clear fixes? :-)

**Comment by TheBlueMatt** [comment]
utACK 6c6494092e57e78bed4b6f2b315ed3e9d328ad9f

File: contrib/verify-commits/verify-commits.sh [hunk 1]
```diff
@@ -38,8 +36,8 @@ PREV_COMMIT=""
 
 while true; do
 	if [ "$CURRENT_COMMIT" = $VERIFIED_ROOT ]; then
-		echo "There is a valid path from "$CURRENT_COMMIT" to $VERIFIED_ROOT where all commits are signed!"
-		exit 0;
+		echo "There is a valid path from \"$CURRENT_COMMIT\" to $VERIFIED_ROOT where all commits are signed!"
```
Comment (MarcoFalke) [inline contrib/verify-commits/verify-commits.sh]: According to the line above this is only printed when both vars are the same. No need to print them both.

**Review by maflcko — COMMENTED** [review]
utACK 6c6494092

File: contrib/verify-commits/verify-commits.sh (hunk 1 above)
Comment (practicalswift) [reply contrib/verify-commits/verify-commits.sh]: @MarcoFalke Do you have any suggestion regarding the correct output to print? :-)

Comment (MarcoFalke) [reply contrib/verify-commits/verify-commits.sh]: maybe

```diff
 PREV_COMMIT=""
+INITIAL_COMMIT=${CURRENT_COMMIT}
 
 while true; do
 	if [ "$CURRENT_COMMIT" = $VERIFIED_ROOT ]; then
-		echo "There is a valid path from "$CURRENT_COMMIT" to $VERIFIED_ROOT where all commits are signed!"
+		echo "There is a valid path from \"$INITIAL_COMMIT\" to $VERIFIED_ROOT where all commits are signed!"

**Comment by practicalswift** [comment]
@MarcoFalke Fixed. Would you mind re-reviewing? :-)

File: contrib/verify-commits/verify-commits.sh [hunk 2]
```diff
@@ -33,10 +33,11 @@ fi
 
 NO_SHA1=1
 PREV_COMMIT=""
+INITIAL_COMMIT=${CURRENT_COMMIT}
```
Comment (TheBlueMatt) [inline contrib/verify-commits/verify-commits.sh]: In quotes, please.

**Comment by practicalswift** [comment]
@TheBlueMatt Fixed! :-)

**Comment by practicalswift** [comment]
Rebased!

**Comment by MarcoFalke** [comment]
Note, you can go ahead and squash fixup commits when you are doing a
rebase anyway.

**Comment by practicalswift** [comment]
@MarcoFalke Good idea! Now squashed. Would you mind re-reviewing? :-)

**Comment by TheBlueMatt** [comment]
utACK 13a81b19df7acff51187655bb755f1371f9c83ca

**Comment by MarcoFalke** [comment]
utACK 13a81b1 (Only looked at the diff, didn't check that the 8 commits do what they advertise)

**Comment by practicalswift** [comment]
Ready for merge? :-)

**Comment by theuni** [comment]
utACK 13a81b19df7acff51187655bb755f1371f9c83ca

**Comment by sipa** [comment]
utACK 13a81b19df7acff51187655bb755f1371f9c83ca

**Pull request merged**